builder/frameworks/espidf.py

```
@@ -998,30 +980,37 @@ def compile_source_files(
 ):
     build_envs = prepare_build_envs(config, default_env, debug_allowed)
     objects = []
-    components_dir = fs.to_unix_path(os.path.join(FRAMEWORK_DIR, "components"))
+    # Canonical, symlink-resolved absolute path of the components directory
+    components_dir_path = (Path(FRAMEWORK_DIR) / "components").resolve()
+    components_dir = str(components_dir_path)
```

Ruff (0.12.2)

Local variable `components_dir` is assigned to but never used

Remove assignment to unused variable `components_dir`

(F841)